jax/lax/lax.py

```diff
@@ -1441,6 +1441,7 @@ def _brcast_to(x, shape):
 ad.defjvp(cos_p, lambda g, x: neg(mul(g, sin(x))))
 
 atan2_p = standard_binop([_float, _float], 'atan2')
+ad.defjvp(atan2_p, lambda g, x, y: -mul(g,y)/(mul(x,x)+mul(y,y)), lambda g, x, y: mul(g,x)/(mul(x,x)+mul(y,y)))
```

Comment:
Instead of writing `mul(x, x)` and `mul(y, y)`, can you change those to `square(x)` and `square(y)`?

Instead of calling `mul(g, y)`, can you write things here like `(g * y)`? Right now these implementations use a mix of infix operators (like `+`, `/`, and `-`) and explicit lax function calls (like `mul`), but I think it's better to use one style or the other. (In this file we've generally followed the convention of calling lax functions directly rather than using infix operators because the infix operators actually correspond to `jax.numpy` operations and so wouldn't make lax technically self-contained. But that makes things harder to write, so let's just use infix operators.)

Also, can you split this onto three lines, like

```python
ad.defjvp(atan2_p,
         lambda g, x, y: ...,
         lambda g, x, y: ...,
```

By the way, a good source for JVP rules (if you don't want to work them out yourself) is [Autograd's numpy_jvps file](https://github.com/HIPS/autograd/blob/e3b525302529d7490769d5c0bcfc7457e24e3b3e/autograd/numpy/numpy_jvps.py#L53-L54).

Reply:
That was really helpful. Thanks for suggested changes. The link to the numpy_jvps actually helped me find a bug in my implementation.

Reply:
Glad to hear it!